app/components/transUnit/TransUnitService.js

```diff
@@ -170,7 +198,26 @@
        updateSaveButton);
 
     function modifyTranslationText(phrase, newText) {
-      phrase.newTranslation = newText;
+      if (phrase.plural) {
+        var transUnitCtrl = controllerList[phrase.id];
+        phrase.newTranslations[transUnitCtrl.focusedTranslationIndex] = newText;
+      } else {
+        phrase.newTranslation = newText;
+      }
+      EventService.emitEvent(EventService.EVENT.TRANSLATION_TEXT_MODIFIED,
+        phrase);
+      EventService.emitEvent(EventService.EVENT.FOCUS_TRANSLATION,
+        phrase);
+    }
+
+    function modifyTranslationsText(phrase, newTexts) {
```

Comment:
I suggest we can remove content and source (singular) in phrase, and only have `contents` and `sources`. This should simplify lots of code as a start. About plural feature, not sure if it can make it in this sprint. Perhaps we can simplify all these in the plural branch.